framework/packages/abstract-client/src/account.rs

```diff
@@ -0,0 +1,198 @@
+    pub(crate) fn new(abstract_account: AbstractAccount<Chain>) -> Self {
+        Self {
+            abstr_account: abstract_account,
+        }
+    }
+
+    pub(crate) fn from_namespace(
+        abstr: &Abstract<Chain>,
+        namespace: String,
+    ) -> AbstractClientResult<Self> {
+        let namespace_response: NamespaceResponse = abstr
+            .version_control
+            .namespace(Namespace::new(&namespace)?)?;
+
+        let abstract_account: AbstractAccount<Chain> =
+            AbstractAccount::new(abstr, Some(namespace_response.account_id));
+
+        Ok(Self::new(abstract_account))
+    }
+
+    pub fn get_account_info(&self) -> AbstractClientResult<AccountInfo<Addr>> {
+        let info_response: InfoResponse = self.abstr_account.manager.info()?;
+        Ok(info_response.info)
+    }
+
+    // Install an application on the account
+    // creates a new sub-account and installs the application on it.
+    pub fn install_app<
+        M: ContractInstance<Chain>
+            + RegisteredModule
+            + InstantiableContract
+            + From<Contract<Chain>>
+            + Clone,
+        C: Serialize,
+    >(
+        &self,
+        configuration: &C,
+        funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        // Create sub account.
+        let sub_account_response = self.abstr_account.manager.create_sub_account(
+            vec![],
+            "Sub Account".to_owned(),
+            None,
+            None,
+            None,
+            None,
+            &[],
+        )?;
+
+        // Unwrap should be fine since we are not expecting this to ever error.
+        let sub_account_id = get_sub_account_id_from_events(sub_account_response.events()).unwrap();
+
+        let sub_account: AbstractAccount<Chain> = AbstractAccount::new(
+            &self.infrastructure()?,
+            Some(AccountId::local(sub_account_id)),
+        );
+
+        let contract = Contract::new(M::module_id().to_owned(), self.environment());
+
+        let app: M = contract.into();
+
+        sub_account.install_app(app.clone(), configuration, Some(funds))?;
+        Ok(Application::new(Account::new(sub_account), app))
+    }
+
+    pub fn admin(&self) -> AbstractClientResult<Addr> {
+        self.abstr_account.manager.address().map_err(Into::into)
+    }
+
+    pub fn proxy(&self) -> AbstractClientResult<Addr> {
+        self.abstr_account.proxy.address().map_err(Into::into)
+    }
+}
+
+fn get_sub_account_id_from_events(events: Vec<Event>) -> Option<u32> {
+    let wasm_abstract_attributes: Vec<Attribute> = events
+        .into_iter()
+        .filter(|e| e.ty == "wasm-abstract")
+        .flat_map(|e| e.attributes)
+        .collect();
+
+    let sub_account_id: Option<u32> = wasm_abstract_attributes
+        .into_iter()
+        .find(|a| a.key == "sub_account_added")
+        .map(|a| a.value.parse().unwrap());
+
+    sub_account_id
+}
+
+pub struct InterchainAccount {}
```

Comment:
👀

Reply:
Haha I think this was some of the starter code you added initially. I haven't though much about how to handle ibc related things here but it should be possible, what do you think?